### DISCUSSION TITLE
Suggesting Features for the Future (Really Ambitious List)

### DISCUSSION THREAD

**Discussion by douchebagmccree** [ryohey/signal#254] [Ideas]
I'm listing down a bunch of features that may or may not be added in the future. It's not all I have, but here it goes...

**Extended Features**
1. You can undo and redo anything you've done, like a mistake. Simply click on one or two buttons with arrow symbols and you're all set to try again.
2. If a midi isn't synced to the BPM, then there's an option to fix that (or two). Simply click once and the editor will detect the song's BPM or the other option where the song gets locked to its actual BPM.
3. You can chose the volume, velocity, pitch, and more for separate notes instead of it all being one line (it's an optional feature). Changing the meter for either settings can be applied with draggable straight lines, sinusoid beizer curving, etc.
4. You can choose different built-in soundfonts for your midi engine (It may also include the classic GS Wavetable Synth soundfont and maybe even the OPL-3 Adlib Sound if that can be appliable).
5. Easier to understand and easier to manage UI (Tempo changer is more easier to use rather than a line as an option), and also Graphical Themes while you're at it (I dunno, seems like something I wanna try).
6. You can invert notes, reverse notes, select multiple tracks, and select all notes in a midi (If you're willing to transpose your song to an entirely different key).

Also, if anyone's willing to add General Standard support for Signal, that'd be great. Midi editors rarely even have General Standard.

Thanks.

-Sincerely, an autistic internet user named DouchebagMcCree

**Comment by AceLuan54** [comment]
Yes! Btw I am also autistic lol

**Reply by VazioFNF** [reply]
I really want to see different themes added to Signal, and not just light mode and dark mode.
